From e0ebf01ef534ae28f09d2fb55477cb1c1bc1a64d Mon Sep 17 00:00:00 2001
From: Fernando Oliveira <fernandoaju78@gmail.com>
Date: Sun, 3 Mar 2024 21:55:30 -0300
Subject: [PATCH] partition: Exclude ksu modules file from userdata backup

Change-Id: I4b42f52ed9ef00e0c9c39b1de80bf47a2ade65cd
Signed-off-by: Fernando Oliveira <fernandoaju78@gmail.com>
(cherry picked from commit eb439c092b81dd11ab11185b273437b79d1f8230)
(cherry picked from commit 918279a4201fb9a1d09cd40e3432df49438d5ff9)
---
 partition.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/partition.cpp b/partition.cpp
index c78e10b9..d7465778 100755
--- a/partition.cpp
+++ b/partition.cpp
@@ -1267,6 +1267,7 @@ void TWPartition::Setup_Data_Media() {
         backup_exclusions.add_absolute_dir("/data/misc/apexdata/com.android.art"); // exclude this dir to prevent "error 255" on AOSP Android 12
 		backup_exclusions.add_absolute_dir("/data/extm"); //exclude this dir to prevent "error 255" on MIUI
 		backup_exclusions.add_absolute_dir("/data/gsi"); // Contains huge files (DSU System image + Userdata image), and won't work after restoration (requires configuration files in metadata)
+		backup_exclusions.add_absolute_dir("/data/adb/ksu/modules.img"); //After ksu 0.8.x the modules.img file became 1tb, which is inhibiting the execution of backups
 		wipe_exclusions.add_absolute_dir(Mount_Point + "/misc/vold"); // adopted storage keys
 		ExcludeAll(Mount_Point + "/system/storage.xml");
 
-- 
2.53.0

